app/Controllers/configureController.php

```diff
@@ -270,7 +270,7 @@ public function archivingAction(): void {
 			];
 		if (!empty(FreshRSS_Context::userConf()->archiving['keep_period'])) {
 			$keepPeriod = FreshRSS_Context::userConf()->archiving['keep_period'];
-			if (preg_match('/^PT?(?P<count>\d+)[YMWDH]$/', $keepPeriod, $matches)) {
+			if (preg_match('/^PT?(?P<count>\d+)[YMWDH]$/', (string) $keepPeriod, $matches)) {
```

Comment:
In the different places where you have used some manual type casting, please try to see whether it could be typed higher up, to avoid the need of manual casting.
And if you do need a type cast, please without space `(string)$s`

Reply:
I suggest you move the type casts to another PR, so this PR can remain clean and focused on syntax